pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/DnsResolverUtil.java

```diff
@@ -19,12 +19,19 @@
 package org.apache.pulsar.common.util.netty;
 
 import io.netty.resolver.dns.DnsNameResolverBuilder;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
+import java.security.Security;
+import java.util.Optional;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class DnsResolverUtil {
+
+    private static final String CACHE_POLICY_PROP = "networkaddress.cache.ttl";
+    private static final String CACHE_POLICY_PROP_FALLBACK = "sun.net.inetaddr.ttl";
+    private static final String NEGATIVE_CACHE_POLICY_PROP = "networkaddress.cache.negative.ttl";
+    private static final String NEGATIVE_CACHE_POLICY_PROP_FALLBACK = "sun.net.inetaddr.negative.ttl";
+    /* default value for positive lookups */
+    private static final int JDK_DEFAULT_TTL = 30;
```

Comment:
I'm in the understanding that the JDK default is infinite for positive lookups. Do you have a reference to share that this is 30 seconds?

Reply:
I've added another comment whether using `JDK_DEFAULT_TTL` is necessary at all. If not, let's remove it.

Reply:
After examining the logic of `InetAddressCachePolicy`, I discovered that it sets a default TTL within the class. If the `networkaddress.cache.ttl` or `sun.net.inetaddr.ttl` parameter is not retrieved, it defaults to a value of 30. This contradicts their documentation.

```
    static {
        Integer tmp = getProperty(cachePolicyProp, cachePolicyPropFallback);
        if (tmp != null) {
            cachePolicy = tmp < 0 ? FOREVER : tmp;
            propertySet = true;
        } else {
            /* No properties defined for positive caching. If there is no
             * security manager then use the default positive cache value.
             */
            if (System.getSecurityManager() == null) {
                cachePolicy = DEFAULT_POSITIVE;
            }
        }
    }
```
[InetAddressCachePolicy.java#L103C5-L130C6](https://github.com/openjdk/jdk/blob/bc73963974a824d77d54b8b0edbf8f05262b721c/src/java.base/share/classes/sun/net/InetAddressCachePolicy.java#L103C5-L130C6)

Reply:
Thanks for explaining @jianyun8023 . Please add a comment to `JDK_DEFAULT_TTL` that this value matches `sun.net.InetAddressCachePolicy.DEFAULT_POSITIVE` which is used when no security manager is used.

Reply:
btw. security manager is deprecated since Java 17 and will be removed in the future.
https://openjdk.org/jeps/411

Reply:
ok

Reply:
This is really giving me a headache!